platooning_strategic/src/state_machine.cpp

```diff
@@ -106,8 +112,29 @@ namespace platoon_strategic
         
     }
 
-    cav_msgs::Maneuver PlatooningStateMachine::planManeuver(double current_dist, double end_dist, double current_speed, double target_speed, int lane_id, ros::Time current_time){
+    cav_msgs::Maneuver PlatooningStateMachine::composeManeuver(){
+        cav_msgs::Maneuver maneuver_msg;
+        
+
+        if (current_platoon_state == PlatoonState::FOLLOWER){
+            ros::Time current_time = ros::Time::now();
+            maneuver_msg.type = cav_msgs::Maneuver::LANE_FOLLOWING;
+            double start_dist = pm_.getCurrentDowntrackDistance();
+            maneuver_msg.lane_following_maneuver.parameters.neogition_type = cav_msgs::ManeuverParameters::PLATOONING;
+            maneuver_msg.lane_following_maneuver.parameters.presence_vector = cav_msgs::ManeuverParameters::HAS_TACTICAL_PLUGIN;
+            maneuver_msg.lane_following_maneuver.parameters.planning_tactical_plugin = "PlatooningTacticalPlugin";
+            maneuver_msg.lane_following_maneuver.parameters.planning_strategic_plugin = "PlatooningStrategicPlugin";
+            maneuver_msg.lane_following_maneuver.start_dist = start_dist;
+            maneuver_msg.lane_following_maneuver.start_speed = pm_.command_speed_;
+            maneuver_msg.lane_following_maneuver.start_time = current_time;
+            maneuver_msg.lane_following_maneuver.end_dist = start_dist + (mvr_duration_*pm_.command_speed_);
+            maneuver_msg.lane_following_maneuver.end_speed = pm_.command_speed_;
+            // because it is a rough plan, assume vehicle can always reach to the target speed in a lanelet
+            maneuver_msg.lane_following_maneuver.end_time = current_time + ros::Duration(mvr_duration_);
+            maneuver_msg.lane_following_maneuver.lane_id = "";
```

Comment:
I'd recommend adding a "//TODO: Populate actual lane ID" comment here, so we don't lose track of this.